diesel/src/mysql/connection/bind.rs

```diff
@@ -0,0 +1,74 @@
+extern crate mysqlclient_sys as ffi;
+
+use mysql::MysqlType;
+use std::mem;
+use std::os::{raw as libc};
+
+pub struct Binds {
+    data: Vec<Option<Vec<u8>>>,
+    lengths: Vec<libc::c_ulong>,
+    is_nulls: Vec<ffi::my_bool>,
+    mysql_binds: Vec<ffi::MYSQL_BIND>,
+}
+
+impl Binds {
+    pub fn from_input_data(input: Vec<(MysqlType, Option<Vec<u8>>)>) -> Self {
+        let lengths = input.iter().map(|x| match *x {
+            (_, Some(ref data)) => data.len() as libc::c_ulong,
+            (_, None) => 0,
+        }).collect();
+        let is_nulls = input.iter()
+            .map(|&(_, ref data)| if data.is_none() { 1 } else { 0 })
+            .collect();
+        let (types, data) = input.into_iter().unzip::<_, _, Vec<_>, _>();
+
+        let mysql_binds = types.into_iter().map(|tpe| {
+            let mut bind: ffi::MYSQL_BIND = unsafe { mem::zeroed() };
+            bind.buffer_type = mysql_type_to_ffi_type(tpe);
+            bind
+        }).collect();
+
+        let mut res = Binds {
+            data: data,
+            lengths: lengths,
+            is_nulls: is_nulls,
+            mysql_binds: mysql_binds,
+        };
+        unsafe { res.link_mysql_bind_pointers(); }
+        res
+    }
+
+    pub fn mysql_binds(&mut self) -> &mut [ffi::MYSQL_BIND] {
+        &mut self.mysql_binds
+    }
+
+    // This function relies on the invariant that no further mutations to this
+    // struct will occur after this function has been called.
+    unsafe fn link_mysql_bind_pointers(&mut self) {
+        for (i, bind_data) in self.data.iter_mut().enumerate() {
+            if let Some(ref mut data) = bind_data.as_mut() {
+                self.mysql_binds[i].buffer = data.as_mut_ptr() as *mut libc::c_void;
+                self.mysql_binds[i].buffer_length = data.capacity() as libc::c_ulong;
+            }
```

Comment:
`None` means that the bind parameter is going to be populated with SQL `NULL`. We're not dealing with `mysql_stmt_bind_result` yet, just `mysql_stmt_bind_param`.

Since the value of the buffer is `None`, `self.is_nulls[i]` will be `1`, and `self.lengths[i]` will be `0`. The value of `buffer` and `buffer_length` are both 0 since we initialized the struct with zeroed memory (I don't think any of that matters though as long as `self.is_nulls[i]` is non-zero)